api/envoy/type/v3/ratelimit_strategy.proto

```diff
@@ -0,0 +1,85 @@
+syntax = "proto3";
+
+package envoy.type.v3;
+
+import "envoy/type/v3/ratelimit_unit.proto";
+import "envoy/type/v3/token_bucket.proto";
+
+import "xds/annotations/v3/status.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.type.v3";
+option java_outer_classname = "RatelimitStrategyProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/type/v3;typev3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+option (xds.annotations.v3.file_status).work_in_progress = true;
+
+// [#not-implemented-hide:]
+// [#protodoc-title: Rate Limit Strategies]
+
+message RateLimitStrategy {
+  // Choose between allow all and deny all.
+  enum BlanketRule {
+    ALLOW_ALL = 0;
+    DENY_ALL = 1;
+  }
+
+  // Limit the number of requests per second.
+  message RequestsPerTimeUnit {
+    // The number of requests per unit of time to allow. Requests over the limit
+    // are denied.
+    // If set to 0, deny all (equivalent to BlanketRule.DENY_ALL).
+    uint64 requests_per_time_unit = 1;
+
+    // The unit of time. Ignored when requests_per_time_unit is 0 (deny all).
+    RateLimitUnit time_unit = 2;
```

Comment:
Should we use Duration here?

Reply:
Not sure how it fits here. This specifies "the number of request" and "per what unit of time", like 10 requests per `RateLimitUnit.SECOND`. If I were to replace it with `Duration`, how would I do it?

Reply:
It looks like we are using the RateLimitUnit in the original rate limit config. We can keep it here as well.